packages/eslint/src/plugins/plugin.ts

```diff
@@ -28,48 +29,72 @@ export const createNodes: CreateNodes<EslintPluginOptions> = [
     baseEsLintConfigFile,
     baseEsLintFlatConfigFile,
   ]),
-  (configFilePath, options, context) => {
+  async (configFilePath, options, context) => {
     options = normalizeOptions(options);
 
     // Ensure that configFiles are set, e2e-run fails due to them being undefined in CI (does not occur locally)
     // TODO(JamesHenry): Further troubleshoot this in CI
     (context as any).configFiles = context.configFiles ?? [];
 
-    // Create a Set of all the directories containing eslint configs
-    const eslintRoots = new Set(context.configFiles.map(dirname));
+    // Create a Set of all the directories containing eslint configs, and a
+    // list of globs to exclude from child projects
+    const eslintRoots = new Set();
+    const filesToExclude: string[] = [];
+    for (const configFile of context.configFiles) {
+      const configDir = dirname(configFile);
+      eslintRoots.add(configDir);
+      if (configDir !== '.' && configDir !== dirname(configFilePath)) {
+        filesToExclude.push(`${configDir}/**/*`);
+      }
```

Comment:
I believe there's a possibility where a parent `configDir` of the `configFilePath` will get added to the `filesToExclude` paths. This would ignore everything including the files in the directory as well.

If `configFiles` is for `libs/a/.eslintrc.json`

```ts
[
  `.eslintrc.json`, // This isn't added because it's the root...
  `libs/.eslintrc.json`, // This one does get added...
  `libs/a/`.eslintrc.json`, // This is the actual config file.
]
```

Reply:
Good catch! I only need to ignore the nested ones. I'll amend it and add tests to cover that.